build/lib/watch/index.js

```diff
@@ -17,9 +17,13 @@ function handleDeletions() {
 	});
 }
 
-const watch = process.platform === 'win32'
-	? require('./watch-win32')
-	: require('gulp-watch');
+const disableNewWatcher = process.env['vscode.disableNewWatcher'];
```

Comment:
Might be better to not have a `.` in the env variable name, we use `VSCODE_VARNAME` format elsewhere afaik.